tests/jobs/test_local_task_job.py

```diff
@@ -372,16 +361,17 @@ def test_localtaskjob_maintain_heart_rate(self):
         # loop in _execute()
         return_codes = [None, 0]
 
-        def multi_return_code():
-            return return_codes.pop(0)
+        def multi_return_code(*args, **kwargs):
+            while return_codes:
+                return return_codes.pop(0)
 
         time_start = time.time()
-        with patch.object(StandardTaskRunner, 'start', return_value=None) as mock_start:
-            with patch.object(StandardTaskRunner, 'return_code') as mock_ret_code:
-                mock_ret_code.side_effect = multi_return_code
-                job1.run()
-                assert mock_start.call_count == 1
-                assert mock_ret_code.call_count == 2
+
+        mock_return_code.side_effect = multi_return_code
```

Comment:
```suggestion
        mock_return_code.side_effect = [None, 0]
```

[Documentation on this](https://docs.python.org/3/library/unittest.mock.html#quick-guide)

Reply:
There's a problem here which I'm trying to understand. From the perspective of the original test, heartbeat should stop beating once the return code is `0` but now, it doesn't stop which is the cause of several issues we have around this.
If we return `[None, 0]`, the heartbeat will still beat for the third time. That was why I added `while return_codes`.

Reply:
For this, I'm experimenting `not heartbeating` whenever return_code is zero and if it works I will correct this test. Should I leave it as quarantined? @ashb @uranusjr

Reply:
> From the perspective of the original test, heartbeat should stop beating once the return code is 0 but now, it doesn't stop which is the cause of several issues we have around this.

Ohhhh… hmm. Not really sure what best to do here.

Reply:
I have found the culprit, here's a PR https://github.com/apache/airflow/pull/17654

Reply:
Turns out this is not really heart beating more but return_code called extra time

Reply:
So since we’ve established `return_code` is called one more time after it exits, would `mock_return_code.side_effect = [None, 0, 0]` work, or can `return_code` be called more than one extra time?

Reply:
I will check now

Reply:
It works with any extra value, I used None and added some explanation. Take a look

Reply:
Yeah because the extra call just goes unused.